tensorflow/core/grappler/optimizers/constant_folding.cc

```diff
@@ -2460,9 +2468,33 @@ bool ConstantFolding::IsReductionSimplifiableToIdentity(
   return simplifiable;
 }
 
+bool ConstantFolding::ReplaceReductionWithIdentity(NodeDef* node) const {
+  // Replace the reduction node with an identity node, that can be further
+  // optimized by the model pruner.
```

Comment:
maybe remove "by the model pruner". There are various optimizers that target Identity, most notably in the dependency optimizer.

Reply:
Done.